mp/src/game/client/neo/neo_fixup_glshaders.cpp

```diff
@@ -0,0 +1,48 @@
+#include "neo_fixup_glshaders.h"
+#ifdef LINUX
+
+// Engine
+#include "filesystem.h"
+#include "icvar.h"
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+// NEO HACK (Rain): This is a hack around a vision shaders corruption on Linux.
+// For details, see: https://github.com/NeotokyoRebuild/neo/pull/587
+void FixupGlShaders(IFileSystem* filesystem, ICvar* cvar)
+{
+	constexpr char filename[] = "glshaders.cfg";
+	constexpr auto pathID = "MOD";
+
+	// This prevents the game from generating the problematic file on exit.
+	// There's also a "mat_autoload_glshaders", but it seems by the time we load,
+	// it's too late for that to take an effect, at least via this route.
+	// The client can still override this by including a command in their autoexec,
+	// if they really want to.
+	auto autosaveGlShaders = cvar->FindVar("mat_autosave_glshaders");
+	if (autosaveGlShaders != nullptr)
+	{
+		autosaveGlShaders->SetValue(false);
+		Assert(!autosaveGlShaders->GetBool());
+	}
```

Comment:
No `else` branch with error logging (which always helps with troubleshooting the bug reports)? Same question for `if` statement on line 41.